### PR TITLE
feat: implement get authRequest by id

### PR DESCRIPTION
**Pull request by muhlemmer**
This PR implement `GetAuthRequest` for the v2 OIDC API.

Note for review: it appears that the `V2_` prefix does not end up in the database. This means that on incoming requests the prefix has to be stripped and in the response added. It might be better to store the complete `V2_<id>` in the database. (TBD).

Related to #5468 

### Definition of Ready

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [x] PR is linked to the corresponding user story
- [x] Acceptance criteria are met
- [x] All open todos and follow ups are defined in a new ticket and justified
- [x] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [x] No debug or dead code
- [x] My code has no repetitions
- [x] Critical parts are tested automatically
- [x] Where possible E2E tests are implemented
- [x] Documentation/examples are up-to-date
- [x] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

File: internal/query/embed/auth_request_by_id.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    client_id,
+    scope,
+    redirect_uri,
+    prompt,
+    ui_locales,
+    login_hint,
+    max_age,
+    hint_user_id
+from projections.auth_requests %s
+where id = $1
```
Comment (livio-a) [inline internal/query/embed/auth_request_by_id.sql:13]:
```suggestion
where id = $1 and instance_id = $2
```

File: internal/query/auth_request.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	}
+
+	var (
+		scope   database.StringArray
+		prompt  database.EnumArray[domain.Prompt]
+		locales database.StringArray
+	)
+
+	dst := new(AuthRequest)
+	err = q.client.DB.QueryRowContext(ctx, q.authRequestByIDQuery(ctx), id).Scan(
```
Comment (livio-a) [inline internal/query/auth_request.go:54]: although the id is unique we should make sure a user can only query his own instance using `authz.GetInstance(ctx).InstanceID()`

File: cmd/defaults.yaml
Comment (livio-a) [inline cmd/defaults.yaml:1]: i'd only give grant to the IAM roles and i think only the (login) client which created the auth request should probably be able to update it
maybe we can have a short discussion about the permissions

Comment (muhlemmer) [reply cmd/defaults.yaml:1]: As discussed, I'm now checking if the `login_client` matches the user ID in the request token. I don't see an onvious way of specifically allowing the IAM_OWNER as well.

**Review by livio-a — APPROVED** [review]

**Review by livio-a — APPROVED** [review]

**Pull request merged**